src/gui/qgsmapcanvas.h

```diff
@@ -236,6 +237,11 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
       @param layer optionally specify different than current layer */
     void zoomToSelected( QgsVectorLayer* layer = nullptr );
 
+    /** Set canvas extent to the bounding box of a feature
+        @param layer the vector layer
+        @param id the feature id*/
+    void zoomToFeatureId( QgsVectorLayer* layer, QgsFeatureId id );
```

Comment:
How about changing this to
```c++
void zoomToFeatureIds( QgsVectorLayer* layer, QgsFeatureIds ids );
```
thus allowing to zoom to the bounding box of one or more arbitrary features?

While it is not necessary for this PR it would add flexibility for the future, e.g. for plugin developers.